granary/meetup.py

```diff
@@ -0,0 +1,102 @@
+# coding=utf-8
+"""Meetup.com source class.
+"""
+
+from . import source
+import logging
+from oauth_dropins.webutil import util
+import urllib.parse, urllib.request
+
+API_BASE = 'https://api.meetup.com'
+API_RSVPS = '/%(urlname)s/events/%(event_id)s/rsvps'
+
+class Meetup(source.Source):
+
+    DOMAIN = 'meetup.com'
+    NAME = 'Meetup.com'
+
+    def __init__(self, access_token):
+        self.access_token = access_token
+        pass
+
+    def create(self, obj, preview=False, include_link=source.OMIT_LINK, ignore_formatting=False):
+        return self._create(obj, preview, include_link, ignore_formatting)
+
+    def post_rsvp(self, urlname, event_id, response):
+        url = API_BASE + API_RSVPS % {
+                'urlname': urlname,
+                'event_id': event_id,
+                }
+        params = 'response=%(response)s' % {
+                'response': response,
+                }
+        logging.debug('Creating RSVP=%(rsvp)s for %(urlname)s %(event_id)s' % {
+            'rsvp': response,
+            'urlname': urlname,
+            'event_id': event_id,
+            })
+        return util.urlopen(urllib.request.Request(
+            url,
+            data=params,
+            headers={
+                'Authorization': 'Bearer %(access_token)s' % {'access_token': self.access_token},
+                }
+            ))
+
+    def _create(self, obj, preview=False, include_link=source.OMIT_LINK, ignore_formatting=False):
+        if not preview in (False, True):
+            return self.return_error('Invalid Preview parameter, must be True or False')
+        verb = obj.get('verb')
+        response = None
+        if 'rsvp-yes' == verb:
+            response = 'yes'
+        elif 'rsvp-no' == verb:
+            response = 'no'
+        elif 'rsvp-maybe' == verb or 'rsvp-interested' == verb:
+            return self.return_error('Meetup.com does not support %(verb)s' % {'verb': verb})
+        else:
+            return self.return_error('Meetup.com syndication does not support %(verb)s' % {'verb': verb})
+
+        event_url = obj.get('inReplyTo')
+        if not event_url:
+            return self.return_error('missing an in-reply-to')
+
+        parsed_url_path = urllib.parse.urlparse(event_url).path
```

Comment:
minor: feel free to use util.UrlCanonicalizer here through line 84 if it helps.

Reply:
Nice, thanks - didn't see that before